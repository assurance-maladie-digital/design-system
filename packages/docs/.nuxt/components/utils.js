// nuxt/nuxt.js#8607
export function wrapFunctional(options) {
  if (!options || !options.functional) {
    return options
  }

  const propKeys = Array.isArray(options.props) ? options.props : Object.keys(options.props || {})

  return {
    render(h) {
      const attrs = {}
      const props = {}

      for (const key in this.$attrs) {
        if (propKeys.includes(key)) {
          props[key] = this.$attrs[key]
        } else {
          attrs[key] = this.$attrs[key]
        }
      }

      return h(options, {
        on: this.$listeners,
        attrs,
        props,
        scopedSlots: this.$scopedSlots,
      }, this.$slots.default)
    }
  }
}
