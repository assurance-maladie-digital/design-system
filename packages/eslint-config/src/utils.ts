// eslint-disable-next-line ts/no-explicit-any
export function renameRules(rules: Record<string, any>, from: string, to: string) {
	return Object.fromEntries(Object.entries(rules)
		.map(([key, value]) => {
			if (key.startsWith(from)) {
				return [to + key.slice(from.length), value];
			}

			return [key, value];
		}));
}
