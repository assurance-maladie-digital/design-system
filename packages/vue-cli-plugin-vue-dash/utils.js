const path = require('path');

/**
 * Capitalize the first letter of a string
 *
 * @param {string} string The string to capitalize
 * @returns {string} The capitalized string
 */
function capitalizeFirstLetter(string) {
	return string.charAt(0).toUpperCase() + string.slice(1);
}

/**
 * Get path relative to current working directory
 *
 * @param {string} value The path to join
 * @returns {string} The normalized path
 */
function getPath(value) {
	return path.join(process.cwd(), value);
}

module.exports = {
	capitalizeFirstLetter,
	getPath
};
