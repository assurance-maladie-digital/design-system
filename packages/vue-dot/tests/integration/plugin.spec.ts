import Vue from 'vue';

import VueDot from '../../src';

import { getComponents, getDirectives } from './utils';

describe('plugin', () => {
	it('should register components & directives', () => {
		Vue.use(VueDot);

		const parts = {
			components: getComponents(Vue),
			directives: getDirectives(Vue)
		};

		expect(parts).toMatchSnapshot();
	});
});
