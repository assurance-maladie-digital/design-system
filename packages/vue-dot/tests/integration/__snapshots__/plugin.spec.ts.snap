// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin should register components & directives 1`] = `
Object {
  "components": Array [
    "CopyBtn",
    "CustomIcon",
    "DataList",
    "DataListItem",
    "DatePicker",
    "FileList",
    "FileUpload",
    "FormBuilder",
    "FormField",
    "HeaderLoading",
    "LangBtn",
    "NotificationBar",
    "PageCard",
    "PaginatedTable",
    "SubHeader",
    "UploadWorkflow",
  ],
  "directives": Array [
    "debounce",
  ],
}
`;
